napari/utils/menus.py

```diff
@@ -0,0 +1,60 @@
+from typing import Callable, List, Optional
+
+from pydantic import Field
+
+from .events import EventedModel
+
+
+class MenuItem(EventedModel):
+    label: str = Field(..., description='Text of the menu item.')
+    id: str = Field(
+        ...,
+        description='Internal identifier of the menu item, to be used when accessing programmatically.',
+    )
+    description: Optional[str] = Field(
+        description='Description to be shown in the status bar when hovered.'
+    )
+    # TODO: add validation to keybinding
+    keybinding: Optional[str] = Field(
```

Comment:
I'm not entirely sure the menu item needs/wants a keybinding.  The _action_ is the thing that's associated with a keybinding (that action may _appear_ in a menu of course, but the linking of a keybinding to a given action/command is independent of that action's appearance in a menu)

Reply:
this is just an idea i stole from blender - each menu can be associated with a keybinding so for example you could do `Shift-A` to open a submenu for adding a layer and then `P` to select points